File: problem_four/4.rb
def do_magic(min, max)
  result = 0
  [*min..max].each do |a|
    [*min..max].each do |b|
      product = a * b
      result = product if padlindromic?(product) && product > result
    end
  end
  result
end

def padlindromic?(number)
  num_string = number.to_s
  num_string == num_string.reverse
end

p do_magic(100, 999)

File: problem_four/4.js
function padlindromic (number) {
  var numberString = number.toString();
  var sortedNumberString = sortString(numberString);
  return numberString === sortedNumberString;
}

function sortString (inputString) {
  return inputString.split("").reverse().join("");
}

(function doMagic (min, max) {
  var result = 0;
  for (var i=min; i<=max; i++){
    for (var j=min; j<=max; j++){
      var product = i * j;
      if (padlindromic(product) && product >= result) {
        result = product;
      }
    }
  }
  console.log(result);
  return result;
})(100,999);